pkgs/by-name/go/gopeed/package.nix

```
@@ -0,0 +1,80 @@
+{ lib
+, fetchurl
+, stdenv
+, autoPatchelfHook
+, dpkg
+, at-spi2-atk
+, cairo
+, gdk-pixbuf
+, glib
+, gtk3
+, harfbuzz
+, libayatana-appindicator
+, libayatana-indicator
+, libdbusmenu
+, pango
+}:
+stdenv.mkDerivation rec {
+  pname = "gopeed";
+  version = "1.5.7";
+
+  src = fetchurl {
+    url = "https://github.com/GopeedLab/gopeed/releases/download/v${version}/Gopeed-v${version}-linux-amd64.deb";
+    hash = "sha256-VIRxn+iHkJSzlX/RB4DsCHgB4+pGM3rJ2SE5ZAJwGwE=";
+  };
+
+  dontConfigure = true;
+
+  dontBuild = true;
+
+  nativeBuildInputs = [ autoPatchelfHook dpkg ];
+
+  buildInputs = [
+    stdenv.cc.cc
+    at-spi2-atk
+    cairo
+    gdk-pixbuf
+    glib
+    gtk3
+    harfbuzz
+    libayatana-appindicator
+    libayatana-indicator
+    libdbusmenu
+    pango
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+      dpkg-deb -x ${src} ./
+
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv usr $out/usr
+    mv opt $out
+
+    install -Dm644 $out/usr/share/icons/hicolor/scalable/apps/gopeed.svg -t $out/share/pixmaps
+    ln -s $out/opt/gopeed/gopeed $out/bin/gopeed
+
+    runHook postInstall
+  '';
```

needs zenity to open directory browsing dialogs
```suggestion
  installPhase = ''
    runHook preInstall

    mv usr/* opt -t $out

    runHook postInstall
  '';

  fixupPhase = ''
    runHook preFixup

    makeWrapper $out/opt/gopeed/gopeed $out/bin/gopeed \
      --prefix PATH : ${lib.makeBinPath [ gnome.zenity ]}

    runHook postFixup
  '';
```